File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/inport/command/MarkBookingActiveCommand.java
package com.dominikgaller.alpinebooking.booking.core.inport.command;

import java.time.Instant;

/**
 * Command carrying the input data for the MarkBookingActive use case.
 *
 * <p>Contains only primitive and standard-library types — no domain value objects.
 * {@code startedAt} is optional; when null the driver resolves the current time via ClockPort.
 * {@code guideTourId} is an optional correlation identifier linking to the guide tour execution.
 *
 * <p>SDD: See {@code documentation/use-cases/uc06-mark-booking-active.spec.md}, section 2.
 */
public record MarkBookingActiveCommand(
        String bookingId,
        Instant startedAt,
        String guideTourId
) {
}

File: src/main/kotlin/com/example/contractmanagement/booking/core/inport/command/MarkBookingActiveCommand.kt
package com.example.contractmanagement.booking.core.inport.command

import java.time.Instant

/**
 * Input for the MarkBookingActive use case (UC06).
 *
 * @property startedAt when the tour began. Null means "the caller has no authoritative
 *   time", and the driver then reads `ClockPort`. A caller relaying a domain event has
 *   one; a caller triggering the transition directly does not.
 * @property guideTourId opaque correlation id owned by the `guide` context; null when the
 *   activation has no guide tour to correlate with.
 *
 * SDD: see `documentation/use-cases/uc06-mark-booking-active.spec.md` § 2.
 */
data class MarkBookingActiveCommand(
    val bookingId: String,
    val startedAt: Instant?,
    val guideTourId: String?,
)
